### PR TITLE
New checkValue method

### PR DESCRIPTION
**Pull request by theSaintBelial**
I rewrote checkValue method due to HashMap arg, now look how it works (or doesnt work)

### CODE REVIEW

File: ru.bmstu.rk9.rao/src/ru/bmstu/rk9/rao/jvmmodel/VarConstCompiler.xtend [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+			
+			members += varconst.toMethod("getName", typeRef(String)) [
+				visibility = JvmVisibility.PUBLIC
+				final = true
+				body = '''
+					return "«vcQualifiedName»";
+				'''
+			]
+			
+			members += varconst.toMethod("checkValue", typeRef(Boolean)) [
```
Comment (theSaintBelial) [inline ru.bmstu.rk9.rao/src/ru/bmstu/rk9/rao/jvmmodel/VarConstCompiler.xtend:52]: Here new checkValue method (old version is a checkLambda now). I just call it in new method

File: ru.bmstu.rk9.rao/src/ru/bmstu/rk9/rao/jvmmodel/VarConstCompiler.xtend [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+					}
+					body = varconst.lambda.body
+				]
+			}
+			
+			members += varconst.toMethod("getAllDependencies", typeRef(ArrayList, typeRef(String))) [
+				visibility = JvmVisibility.PUBLIC
+				if (varconst.lambda !== null)
+				{
+//					«»
```
Comment (MisterPotz) [inline ru.bmstu.rk9.rao/src/ru/bmstu/rk9/rao/jvmmodel/VarConstCompiler.xtend]: i guess you can remove this comment

File: ru.bmstu.rk9.rao.lib/src/ru/bmstu/rk9/rao/lib/varconst/VarConst.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+public abstract class VarConst {
+	private Double	start;
+	private Double	stop;
+	private Double	step;
+	private LambdaExpression lambda;
+	
+	public abstract String getName();
+	
+//	checkValue method creates in VarConstCompiler.xtend
+//	and getAllDependencies method too
```
Comment (MisterPotz) [inline ru.bmstu.rk9.rao.lib/src/ru/bmstu/rk9/rao/lib/varconst/VarConst.java]: you can remove this comment now 
```suggestion
	public abstract boolean checkValue(HashMap<String, Double> args);
```
and replace it with abstract method

**Review by MisterPotz — COMMENTED** [review]
looks good, just add the abstract method

**Pull request closed without merging**